### PR TITLE
Feature: Added Endpoint to update scheduled command sequence

### PR DESCRIPTION
**Pull request by Umang-Rajkarnikar**
Created an endpoint to update an existing scheduled command sequence
- Set up mongodb mock server when performing jest unit tests
- Added a test to verify the functionality of endpoint
- The other test cases have been described in comments, need to do in separate pr

Related issues: #119

### CODE REVIEW

File: src/leo-server-app/src/database/database.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const mongoose = require("mongoose");
+
+const { MongoMemoryServer } = require("mongodb-memory-server");
+let mongod: any = null;
+
+const connectDB = async (mode?: string) => {
+  //   try {
```
Comment (quinnha) [inline src/leo-server-app/src/database/database.ts:7]: remove comment

File: src/leo-server-app/src/database/database.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  let dbUrl = process.env.DB_URI;
+  console.log(process.env.NODE_ENV);
+  //   if (process.env.NODE_ENV === "test") {
```
Comment (quinnha) [inline src/leo-server-app/src/database/database.ts:10]: see above

File: src/leo-server-app/src/database/database.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    dbUrl = mongod.getUri();
+  }
+
+  const conn = await mongoose.connect(dbUrl);
+
+  console.log(`MongoDB connected: ${conn.connection.host}`);
+  //   } catch (err) {
+  //     console.log(err);
+  //     process.exit(1);
+  //   }
```
Comment (quinnha) [inline src/leo-server-app/src/database/database.ts:22]: remove

File: src/leo-server-app/src/models/command.ts [hunk 4]
```diff
@@ -33,4 +33,4 @@ export const commandSchema = new Schema(
 
 const Command = mongoose.model("Command", commandSchema);
 
-module.exports = Command;
+export default Command;
```
Comment (quinnha) [inline src/leo-server-app/src/models/command.ts:36]: is there a particular reason to change these?

File: src/leo-server-app/src/routes/schedule.ts [hunk 5]
```diff
@@ -80,6 +96,76 @@ router.post("/createSchedule", async (req: CreateScheduleProp, res: any) => {
...
+      {
+        command,
+      },
+      { new: true }
+    ).exec();
+
+    return res.json({ message: "Updated command", updatedCommand });
+  }
+);
+
```
Comment (quinnha) [inline src/leo-server-app/src/routes/schedule.ts:168]: tests should be a ina separate testing file, not sure what its doing here

File: src/leo-server-app/src/tests/schedule.test.ts [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+  let pastSchedule: any;
+  let futureSchedule: any;
+  let commands: any[] = [];
+
+  beforeAll(async () => {
+    console.log(process.env.NODE_ENV);
+    connectDB("test");
+
+    // Create command record
```
Comment (quinnha) [inline src/leo-server-app/src/tests/schedule.test.ts:30]: empty comment?

File: src/leo-server-app/src/tests/schedule.test.ts [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    };
+
+    commands = await Command.create([pastCommand, futureCommand]);
+  });
+
+  afterAll(async () => {
+    disconnectDB();
+  });
+
+  it("Correctly update existing command record", async () => {
```
Comment (quinnha) [inline src/leo-server-app/src/tests/schedule.test.ts:100]: nit: "updates" vs "update"

File: src/leo-server-app/src/tests/schedule.test.ts [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+// Test 1 - update correctly
+
+// Test 2 - update with invalid user credentials
+
+// Test 3 - update as ADMIN
+
+// Test 4 - update with invalid ids
+
+// Test 5 - Update with invalid command sequence
```
Comment (quinnha) [inline src/leo-server-app/src/tests/schedule.test.ts:121]: are you going to add these tests in a different pr?

Comment (Umang-Rajkarnikar) [reply src/leo-server-app/src/tests/schedule.test.ts:121]: yes

File: src/leo-server-app/src/routes/schedule.ts (hunk 5 above)
Comment (Umang-Rajkarnikar) [reply src/leo-server-app/src/routes/schedule.ts:168]: oh mb these are validation checks, just labelled them as tests, lemme add betteer comment

**Review by RishiVaya — APPROVED** [review]
lgtm

File: src/leo-server-app/src/models/command.ts (hunk 4 above)
Comment (Umang-Rajkarnikar) [reply src/leo-server-app/src/models/command.ts:36]: For some reason, the export it like this and importing it using require makes the export lose its type

**Review by quinnha — APPROVED** [review]
lgtm, good stuff

**Pull request merged**